### PR TITLE
[pytorch] Add NDList to IValue unit test

### PR DESCRIPTION
**Pull request by frankfliu**
The testing code can be used as example as well

Change-Id: I693624fd111afb0a5547d619a03d2cb15831af1b

## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: engines/pytorch/pytorch-engine/src/test/java/ai/djl/pytorch/jni/IValueUtilsTest.java [hunk 1]
```diff
@@ -42,4 +42,42 @@ public void getInputsTestTupleSyntax() {
             Arrays.stream(iValues).forEach(IValue::close);
         }
     }
+
+    @Test
+    public void testMapOfTensor() {
+        try (NDManager manager = NDManager.newBaseManager()) {
+            NDArray array1 = manager.zeros(new Shape(1));
+            array1.setName("input1.key1");
```
Comment (ylwu-amzn) [inline engines/pytorch/pytorch-engine/src/test/java/ai/djl/pytorch/jni/IValueUtilsTest.java:50]: minor: add some comments to explain why use `input1` as prefix? Like group all keys with same prefix into one Map?

**Review by zachgk — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for adding these tests, good examples to follow for new DJL users like me. Appreciate it!

File: engines/pytorch/pytorch-engine/src/test/java/ai/djl/pytorch/jni/IValueUtilsTest.java [hunk 2]
```diff
@@ -42,4 +43,54 @@ public void getInputsTestTupleSyntax() {
...
+            NDArray array2 = manager.ones(new Shape(1));
+            array2.setName("input1.key2");
+            NDArray array3 = manager.zeros(new Shape(1));
+            array3.setName("input2.key1");
+            NDArray array4 = manager.ones(new Shape(1));
+            array4.setName("input2.key2");
+            NDArray array5 = manager.ones(new Shape(1));
+            array5.setName("input2.key3");
+            NDList input = new NDList(array1, array2, array3, array4, array5);
+            // the NDList is mapped to (input1: Dict(str, Tensor), input2: Dict(str, Tensor))
```
Comment (ylwu-amzn) [inline engines/pytorch/pytorch-engine/src/test/java/ai/djl/pytorch/jni/IValueUtilsTest.java:61]: Any way to generate some NDList like 

> (input1: Dict(str, **list(Tensor)**), input2: Dict(str, **list(Tensor)**))

Comment (frankfliu) [reply engines/pytorch/pytorch-engine/src/test/java/ai/djl/pytorch/jni/IValueUtilsTest.java:61]: 1. We don't support this currently, it's possible to add nested IValue support.
2. But you have to use IValue directly, and use PyTorch specific API, which doesn't work with Translator pattern. something like: https://github.com/deepjavalibrary/djl/blob/master/engines/pytorch/pytorch-engine/src/test/java/ai/djl/pytorch/integration/IValueTest.java#L184-L191

Comment (frankfliu) [reply engines/pytorch/pytorch-engine/src/test/java/ai/djl/pytorch/jni/IValueUtilsTest.java:61]: I created a PR for your issue: https://github.com/deepjavalibrary/djl/pull/1765
